pkg/sniconn/Readme.md

```diff
@@ -0,0 +1,390 @@
+- The amount of data sent from the source IP address to the SNI in each of the individual HTTP and TLS connections
+- The destination IP addresses and ports of the HTTP and TLS servers responding to the SNI
+- The certificate subjects presented by the TLS servers responding to the SNI
+- The JA3 hashes of the TLS configurations sent from the source IP address to the TLS servers responding to the SNI
+- The JA3S hashes of the TLS configurations presented by the TLS servers responding to the SNI
+- The HTTP methods of the requests sent from the source IP address to the HTTP servers responding to the SNI
+- The HTTP user agents of the requests sent from the source IP address to the HTTP servers responding to the SNI
+
+
+## Package Outputs
+
+## Source Unique IP, Destination SNI Pair
+Inputs:
+- `ParseResults.TLSConnMap` created by `FSImporter`
+    - Field: `Hosts`
+        - Type: data.UniqueSrcFQDNPair
+- `ParseResults.HTTPConnMap` created by `FSImporter`
+    - Field: `Hosts`
+        - Type: data.UniqueSrcFQDNPair
+
+Outputs:
+- MongoDB `SNIconn` collection:
+    - Field: `src`
+        - Type: string
+    - Field: `src_network_uuid`
+        - Type: UUID
+    - Field: `src_network_name`
+        - Type: string
+    - Field: `fqdn`
+        - Type: string
+
+The `src` field records the string representation of the IP address of the source of a SNI connection as seen in the network logs. Similarly, the field `fqdn` specifies the server name indicator/ fully qualified domain name of the destination of a SNI connection as seen in the network logs. The `src_network_uuid` and `src_network_name` fields have been introduced to disambiguate hosts using the same private IP address on separate networks.
+
+These fields are used to select an individual entry in the `SNIconn` collection. All of the other outputs described here use the `src`, `src_network_uuid`, and `fqdn` fields as selectors when updating `SNIconn` collection entries in MongoDB.
+
+### Chunk ID
+Inputs:
+- `Config.S.Rolling.CurrentChunk`
+    - Type: int
+
+Outputs:
+- MongoDB `SNIconn` collection:
+    - Field: `cid`
+        - Type: int
+
+The `cid` field records the chunk ID of the import session in which this unique connection document was last updated. This field is used to support rolling imports.
+
+### TLS Destination IP Addresses and Ports
+Inputs:
+- `ParseResults.TLSConnMap` created by `FSImporter`
+    - Field: `RespondingIPs`
+        - Type: data.UniqueIPSet
+    - Field: `RespondingPorts`
+        - Type: data.IntSet
+
+Outputs:
+- MongoDB `SNIconn` collection:
+    - Array Field: `dat`
+        - Object Field: `tls`
+            - Array Field: `dst_ips`
+                - Field: `ip`
+                    - Type: string
+                - Field: `network_uuid`
+                    - Type: UUID
+                - Field: `network_name`
+                    - Type: string
+            - Array Field: `dst_ports`
+                - Type: int
+            - Field: `cid`
```

Comment:
This highlights a bigger issue with removing outdated cids. Since the cids are located under `dat.tls.cid`, `dat.http.cid`, and `dat.beacon.cid`, the remover doesn't take these out. To fix this, I will move the `cid` fields up alongside the `tls` and `http` entries.

These fields `tls.cid` and `http.cid` will be removed. 

I tested rolling updates, but I forgot to test rolling over to the next day.